src/obs-text-pthread-main.c

```diff
@@ -49,11 +49,16 @@ static void *tp_create(obs_data_t *settings, obs_source_t *source)
 
 	obs_enter_graphics();
 	if (!textalpha_effect) {
-		char *f = obs_module_file("textalpha.effect");
-		textalpha_effect = gs_effect_create_from_file(f, NULL);
-		if (!textalpha_effect)
-			blog(LOG_ERROR, "Cannot load '%s'", f);
-		bfree(f);
+		const char *filename = "textalpha.effect";
+		char *f = obs_module_file(filename);
+		if (f) {
+			textalpha_effect = gs_effect_create_from_file(f, NULL);
+			if (!textalpha_effect)
+				blog(LOG_ERROR, "Cannot load '%s'", f);
+			bfree(f);
+		} else {
+			blog(LOG_ERROR, "Cannot find module file '%s'", filename);
+		}
```

Comment:
I prefer to report the error and exit so that the indent of the main code is not deep.
Can we have a separate static function for this block so that we can return from the function if `obs_module_file` returns NULL?

```c
static void initialize_textalpha_effect()
{
	const char *filename = "textalpha.effect";
	char *f = obs_module_file(filename);
	if (!f) {
		blog(LOG_ERROR, "Cannot find module file '%s'", filename);
		return;
	}

	textalpha_effect = gs_effect_create_from_file(f, NULL);
	if (!textalpha_effect)
		blog(LOG_ERROR, "Cannot load '%s'", f);

	bfree(f);
}
```